src/main/java/com/google/api/codegen/transformer/InitCodeTransformer.java

```diff
@@ -565,10 +589,19 @@ private InitValueView getInitValue(GapicMethodContext context, InitCodeNode item
     return formatFunctionArgs;
   }
 
-  private List<FieldSettingView> getFieldSettings(
+  /** Returns true if fieldConfigs contains fieldConfig. */
+  private boolean hasFieldConfig(Iterable<FieldConfig> fieldConfigs, FieldConfig fieldConfig) {
+    return StreamSupport.stream(fieldConfigs.spliterator(), false)
+        .anyMatch(f -> f.equals(fieldConfig));
+  }
+
+  /** Returns an insertion order map of FieldSettingView to whether or not the field is required. */
+  private LinkedHashMap<FieldSettingView, Boolean> getFieldSettings(
```

Comment:
nit: using a `Map<FieldSettingView, Boolean>` to store whether a field is required is unintuitive for me. Why not just make whether it is required part of the `View`?

Reply:
That's a great point, done.